bot/exts/utilities/conversationstarters.py

```diff
@@ -35,35 +38,75 @@
 class ConvoStarters(commands.Cog):
     """General conversation topics."""
 
-    @commands.command()
-    @whitelist_override(channels=ALL_ALLOWED_CHANNELS)
-    async def topic(self, ctx: commands.Context) -> None:
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @staticmethod
+    def _build_topic_embed(channel_id: int) -> discord.Embed:
         """
-        Responds with a random topic to start a conversation.
+        Build an embed containing a conversation topic.
 
         If in a Python channel, a python-related topic will be given.
-
         Otherwise, a random conversation topic will be received by the user.
         """
         # No matter what, the form will be shown.
-        embed = Embed(description=f"Suggest more topics [here]({SUGGESTION_FORM})!", color=Color.blurple())
+        embed = discord.Embed(
+            description=f"Suggest more topics [here]({SUGGESTION_FORM})!",
+            color=discord.Color.blurple()
+        )
 
         try:
-            # Fetching topics.
-            channel_topics = TOPICS[ctx.channel.id]
-
-        # If the channel isn't Python-related.
+            channel_topics = TOPICS[channel_id]
         except KeyError:
+            # Channel doesn't have any topics.
             embed.title = f"**{next(TOPICS['default'])}**"
-
-        # If the channel ID doesn't have any topics.
         else:
             embed.title = f"**{next(channel_topics)}**"
+        return embed
+
+    def _predicate(self, message: discord.Message, reaction: discord.Reaction, user: discord.User) -> bool:
+        right_reaction = (
+            user != self.bot.user
+            and reaction.message.id == message.id
+            and str(reaction.emoji) == "🔄"
+        )
+        if not right_reaction:
+            return False
+        is_moderator = any(role.id in MODERATION_ROLES for role in getattr(user, "roles", []))
```

Comment:
```suggestion
            return False

        is_moderator = any(role.id in MODERATION_ROLES for role in getattr(user, "roles", []))
```
Mostly a styling issue

Reply:
Force pushed to the PR